phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java

```diff
@@ -412,7 +412,9 @@ public Expression visit(ColumnParseNode node) throws SQLException {
         // If the column is matched in a WHERE clause against a constant not equal to it's constant, then the entire
         // query would become degenerate.
         if (!resolveViewConstants && IndexUtil.getViewConstantValue(column, ptr)) {
-            return LiteralExpression.newConstant(column.getDataType().toObject(ptr), column.getDataType());
+            return LiteralExpression.newConstant(
```

Comment:
Isn't the sortOrder in the third parameter sufficient?

Reply:
The toObject is used to store the value object and the third parameter is used to invert the object value to its byte representation in the LiteralExpression class. Both value and bytevalue are used in LiteralExpression class for different reasons. 
Maybe need to do some due diligence - that it is not getting inverted twice?

Reply:
Exactly, it seems we might not need 2nd argument, 1st one should be sufficient to avoid double invert. Let me confirm that first.
Thanks @ranganathg @jpisaac !!

Reply:
we need 3rd argument to correctly reflect byte[] value